CodeEdit/Commands/CommandPaletteView.swift

```diff
@@ -0,0 +1,285 @@
+    func onKeyDown(with event: NSEvent) -> Bool {
+
+        switch event.keyCode {
+            // down arrow button
+        case 125:
+            selectNext()
+            return true
+            // up arrow button
+        case 126:
+            selectPrev()
+            return true
+            // enter button
+        case 36:
+            if let command = self.selectedItem {
+                callHandler(command: command)
+            }
+            return true
+            // esc button
+        case 53:
+            closePalette()
+            return true
+        default:
+            return false
+        }
+    }
+
+    func onQueryChange(text: String) {
+        state.commandQuery = text
+        state.fetchMatchingCommands(val: text)
+    }
+
+    func onCommandClick(command: Command) {
+        self.selectedItem = command
+        callHandler(command: command)
+    }
+
+    func textColor(command: Command) -> Color {
+        if self.selectedItem == command {
+            return .white
+        }
+
+        return colorScheme == .dark ? .white : .black
+    }
+
+    public var body: some View {
+        VStack(spacing: 0.0) {
+                HStack(alignment: .center, spacing: 0) {
+                    Image(systemName: "command")
+                        .resizable()
+                        .scaledToFit()
+                        .frame(width: 16, height: 16)
+                        .padding(.leading, 20)
+                        .offset(x: 0, y: 1)
+                    ActionAwareInput(onDown: onKeyDown,
+                                     onTextChange: onQueryChange, text: $state.commandQuery)
+                        .font(.system(size: 24, weight: .light, design: .default))
+                        .padding(16)
+                        .frame(height: 52, alignment: .center)
+                        .foregroundColor(Color(.systemGray).opacity(0.85))
+                        .background(EffectView(.sidebar, blendingMode: .behindWindow))
+                }
+
+            Divider()
+            VStack(spacing: 0) {
+                List(commandsList, selection: $state.selected) { command in
+                    // swiftlint:disable multiple_closures_with_trailing_closure
+                    Button(action: { onCommandClick(command: command) }) {
+                        VStack {
+                            SearchResultLabel(labelName: command.title, textToMatch: state.commandQuery,
+                                              fontColor: textColor(command: command))
+                                .padding(EdgeInsets.init(top: 0, leading: 8, bottom: 0, trailing: 0))
+                        }.frame(maxWidth: .infinity, maxHeight: 15, alignment: .leading)
+                    }.frame(maxWidth: .infinity, maxHeight: 15, alignment: .leading)
+                        .listRowInsets(EdgeInsets.init(top: 0, leading: 0, bottom: 0, trailing: 0))
```

Comment:
You can omit the `init` and call the initializer directly:

```swift
EdgeInsets(top: 0, leading: 0, bottom: 0, trailing: 0)
```

or in this case even better:

```swift
EdgeInsets.zero

// or since EdgeInsets will be implicit, even:
.zero
```